### PR TITLE
#737 Isolated RealmSchema and couple of other minor doc changes out

### PR DESCRIPTION
**Pull request by AndyDentFree**
This reduces the noise in the other PR - should be JUST comment changes. Will use this to simplify other primary key PR

@fealebenpae or @kristiandupont to approve

### CODE REVIEW

File: Realm.Shared/RealmObject.cs [hunk 1]
```diff
@@ -567,7 +567,7 @@ protected void SetByteArrayValue(string propertyName, byte[] value)
         /// Compare objects with identity query for persistent objects.
         /// </summary>
         /// <remarks>Persisted RealmObjects map their properties directly to the realm with no caching so multiple instances of a given object always refer to the same store.</remarks>
-        /// <param name="obj"></param>
+        /// <param name="obj">Object being compared against to see if is same C# object or maps to same managed object in Realm.</param>
```
Comment (nirinchev) [inline Realm.Shared/RealmObject.cs]: should be "...see if *it* is *the* same C# object or maps to *the* same..."

File: Realm.Shared/Schema/Property.cs [hunk 2]
```diff
@@ -21,6 +21,9 @@
 
 namespace Realms.Schema
 {
+    /// <summary>
+    /// Describe a single property of a class stored in a Realm.
```
Comment (nirinchev) [inline Realm.Shared/Schema/Property.cs]: Probably "Describe*s*"

File: Realm.Shared/Schema/RealmSchema.cs [hunk 3]
```diff
@@ -44,6 +56,12 @@ private RealmSchema(SchemaHandle handle, IEnumerable<ObjectSchema> objects)
             Handle = handle;
         }
 
+        /// <summary>
+        /// Find the definition of a class in this schema.
```
Comment (nirinchev) [inline Realm.Shared/Schema/RealmSchema.cs]: Not sure what the convention at Realm is, but the .NET documentation usually describes what the method does from a 3rd person point of view, e.g. "[It] Finds the definition of a class in this schema."

File: Realm.Shared/Schema/RealmSchema.cs [hunk 4]
```diff
@@ -44,6 +56,12 @@ private RealmSchema(SchemaHandle handle, IEnumerable<ObjectSchema> objects)
...
+        /// </summary>
+        /// <param name="name">A valid class name which may be in this schema.</param>
+        /// <exception cref="ArgumentException">Thrown if an empty name is supplied.</exception>
```
Comment (nirinchev) [inline Realm.Shared/Schema/RealmSchema.cs]: Should this be "null or empty name"?

File: Realm.Shared/linq/RealmResults.cs [hunk 5]
```diff
@@ -148,11 +148,16 @@ IEnumerator IEnumerable.GetEnumerator()
 
 
         /// <summary>
-        /// Fast count all objects of a given class.
+        /// Fast count all objects of a given class, or in a RealmResults after casting.
         /// </summary>
         /// <remarks>
         /// Resolves to this method instead of the LINQ static extension <c>Count&lt;T&gt;(this IEnumerable&lt;T&gt;)</c>, when used directly on Realm.All.
+		/// <br>
```
Comment (nirinchev) [inline Realm.Shared/linq/RealmResults.cs]: Indentation issue

File: Realm.Shared/linq/RealmResults.cs [hunk 6]
```diff
@@ -148,11 +148,16 @@ IEnumerator IEnumerable.GetEnumerator()
...
+		/// if someone CASTS a RealmResults<blah> variable from a Where call to
```
Comment (nirinchev) [inline Realm.Shared/linq/RealmResults.cs]: Should this be `RealmResults&lt;T&gt;`? Or does the `remarks` tag properly handle those? Even if that's the case, we probably should replace `<blah>` with `<T>`

**Comment by AndyDentFree** [comment]
retest this

**Comment by AndyDentFree** [comment]
retest this please

File: Realm.PCL/Schema/ObjectSchemaPCL.cs [hunk 7]
```diff
@@ -48,21 +67,38 @@ public IEnumerator<Property> GetEnumerator()
 
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
 
+        /// <summary>
+        /// Creates a schema describing .
```
Comment (kristiandupont) [inline Realm.PCL/Schema/ObjectSchemaPCL.cs]: Is there a word missing here?

File: Realm.PCL/Schema/RealmSchemaPCL.cs [hunk 8]
```diff
@@ -22,21 +22,43 @@
 
 namespace Realms
 {
+    /// <summary>
+    /// Describes the complete set of classes which may be stored in a Realm, either from assembly declarations or, dynamically, by evaluating a Realm from disk.
+    /// </summary>
+    /// <remarks>
+    /// By default this will be all the RealmObjects in all your assemblies unless you restrict with RealmConfiguration.ObjectClasses. 
+    /// Just because a given class <em>may</em> be stored in a Realm doesn't imply much overhead. There will be a small amount of metadata
```
Comment (kristiandupont) [inline Realm.PCL/Schema/RealmSchemaPCL.cs]: This doesn't seem like the right place for this bit of information to me. We could add it to the FAQ if necessary, or in some general documentation somewhere.

**Comment by kristiandupont** [comment]
:+1:

File: Realm.PCL/Schema/RealmSchemaPCL.cs (hunk 8 above)
Comment (AndyDentFree) [inline Realm.PCL/Schema/RealmSchemaPCL.cs]: Our docs seem very lean compared to the other SDKs

Comment (kristiandupont) [inline Realm.PCL/Schema/RealmSchemaPCL.cs]: Sure, I just mean that this is not a place where you would look for this information. The overhead concern is not limited to people who define specific schemas and you are unlikely to encounter this class if you don't (even then it's unlikely as you just need `RealmConfiguration`).

**Pull request merged**